R/DiseasyModel.R

```diff
@@ -0,0 +1,200 @@
+                        checkmate::check_class(activity, "DiseasyActivity", null.ok = TRUE),
+                        add = coll)
+      checkmate::assert(checkmate::check_logical(observables, null.ok = TRUE),
+                        checkmate::check_class(observables, "DiseasyObservables", null.ok = TRUE),
+                        add = coll)
+      checkmate::assert(checkmate::check_logical(season, null.ok = TRUE),
+                        checkmate::check_class(season, "DiseasySeason", null.ok = TRUE),
+                        add = coll)
+      checkmate::assert_character(label, len = 1, any.missing = FALSE, null.ok = TRUE, add = coll)
+      checkmate::reportAssertions(coll)
+
+      # Pass further arguments to the DiseasyBaseModule initializer
+      super$initialize(...)
+
+      # Then try to set the modules
+      if (isTRUE(observables)) {
+        self$load_module(DiseasyObservables$new())
+      } else if (inherits(observables, "DiseasyObservables")) {
+        self$load_module(observables)
+      }
+
+      if (isTRUE(activity)) {
+        self$load_module(DiseasyActivity$new())
+      } else if (inherits(activity, "DiseasyActivity")) {
+        self$load_module(activity)
+      }
+
+      if (isTRUE(season)) {
+        self$load_module(DiseasySeason$new())
+      } else if (inherits(season, "DiseasySeason")) {
+        self$load_module(season)
+      }
+
+      # Set the label for the model
+      private$label <- label
+    },
+
+
+    # Roxygen has only limited support for R6 docs currently, so we need to do some tricks for the documentation
+    # of get_results
+    #' @description `r rd_get_results_description`
+    #' @param observable `r rd_observable()`
+    #' @param prediction_length `r rd_prediction_length()`
+    #' @param quantiles `r rd_quantiles()`
+    #' @param stratification `r rd_stratification()`
+    #' @return `r rd_get_results_return`
+    #' @seealso `r rd_get_results_seealso`
+    get_results = function(observable, prediction_length, quantiles = NULL, stratification = NULL) {
+      private$not_implemented_error("Each model must implement their own `get_results` methods")
+    },
+
+
+    #' @description
+    #'   A method that returns training data for the models based on the model value of `training_length` and
+    #'   the `last_queryable_date` of the `DiseasyObservables` module.
+    #' @param observable `r rd_observable()`
+    #' @param stratification `r rd_stratification()`
+    #' @return The output of `DiseasyObservables$get_observation` constrained to the training period.
+    get_training_data = function(observable, stratification = NULL) {
+
+      # Input validation
+      coll <- checkmate::makeAssertCollection()
+      checkmate::assert_character(observable, add = coll)
+      checkmate::assert_number(self$parameters$training_length, add = coll)
+      checkmate::assert_date(self$observables$last_queryable_date, add = coll)
```

Comment:
Here it is implicit that the unit of time is days. What if a model runs hourly?

Reply:
The model can run hourly with no issue as far as I can tell.

The problem would be that the training data has a resolution of days in the current implementation.

We could make some sweeping changes to allow for a generic time interval as the unit for the observations, but that would be best to include as separate feature request in my opinion since it would touch on many different elements of `diseasy`.